cmd/goProbe/config/monitor.go

```diff
@@ -0,0 +1,112 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/els0r/goProbe/pkg/logging"
+)
+
+const defaultReloadInterval = 5 * time.Minute
+
+// Monitor denotes a config monitor / manager
+type Monitor struct {
```

Comment:
This one basically wraps the config path, loads it and allows both explicit and periodic reloading of the file. The config is stored in the manager (and can be retrieved via its `Config()` method) and is mutex-locked to avoid any races.
During a reload, a callback function can be provided (in this case the call to `CaptureManager.Update()`, which will cross-check for any changes in the config _AND_ attempt to restore the state present in that config, i.e. also bring up interfaces that were torn down before).